File: src/main/java/com/gusrylmubarok/reddit/backend/repository/PostRepository.java
package com.gusrylmubarok.reddit.backend.repository;

import com.gusrylmubarok.reddit.backend.model.Post;
import com.gusrylmubarok.reddit.backend.model.Subreddit;
import com.gusrylmubarok.reddit.backend.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface PostRepository extends JpaRepository<Post, Long> {
    Optional<Post> findAllBySubreddit(Subreddit subreddit);
    Optional<Post> findByUser(User user);
}

File: src/main/kotlin/com/reddot/repository/PostRepository.kt
package com.reddot.repository

import com.reddot.data.entity.Post
import com.reddot.data.entity.Subreddit
import com.reddot.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PostRepository : JpaRepository<Post, Long> {
    fun findAllBySubreddit(subreddit: Subreddit): List<Post>
    fun findByUser(user: User): List<Post>
}